### PR TITLE
ENH: Unify processing

### PR DESCRIPTION
**Pull request by larsoner**
1. Add `params.score = score` so that the scoring function is actually used
2. Modify scoring function to add 100 to oddball, 1000 to AM, and 10000 to IDs so that event numbers don't collide
3. Remove separate YAML files
4. Add all conditions to `badbaby.yml`

I'm currently running bad_101 and bad_102, in theory they should complete and produce reports but we'll see. @ktavabi feel free to ask questions inline and I can respond or we can have a quick Q&A chat / discussion at some point.

There are one or two small fixes in `mnefun` needed, too, but I have to stop for the day, will open a PR tomorrow.

### CODE REVIEW

File: badbaby/processing/badbaby.yml [hunk 1]
```diff
@@ -80,12 +9,12 @@ fetch_raw:
...
       "bad_103",
       "bad_104",
-      bad_105',
+      "bad_105",
       "bad_109",
       "bad_110",
       "bad_113",
       "bad_114",
-      "bad_116a",
+      "bad_116",
```
Comment (larsoner) [inline badbaby/processing/badbaby.yml:17]: Note to self, I should probably revert this...

File: badbaby/processing/badbaby.yml [hunk 2]
```diff
@@ -320,15 +324,17 @@ preprocessing:
...
   filtering:
     lp_cut: 50
     lp_trans: auto
+    hp_cut: 0.5
+    hp_trans: auto
     filter_length: auto
-    fir_design: firwin2
-    phase: zero-double
+    fir_design: firwin
+    phase: zero
```
Comment (ktavabi) [inline badbaby/processing/badbaby.yml:223]: Are these new filtering and higher st_correlation more favorable for infant datasets?

File: badbaby/processing/badbaby.yml [hunk 3]
```diff
@@ -407,13 +413,55 @@ preprocessing:
...
+              [standard, ba, wa],
+              [standard, deviant],
+              [tone],
+              [s1, s2, s3, s4, s5]]
+  out_numbers: [[ 1,  1,  1,  1,  1,  1,  1,  1,  1],
+                [ 1,  2,  3, -1, -1, -1, -1, -1, -1],
+                [ 1,  2,  2, -1, -1, -1, -1, -1, -1],
+                [-1, -1, -1,  1, -1, -1, -1, -1, -1],
+                [-1, -1, -1, -1,  1,  2,  3,  4,  5]]
+  must_match: [[], [], [], []]
```
Comment (ktavabi) [inline badbaby/processing/badbaby.yml]: shouldn't `len(must_match) ==  len(out_names)?

File: badbaby/processing/badbaby.yml [hunk 4]
```diff
@@ -407,13 +413,55 @@ preprocessing:
...
+  analyses: [All, Individual, Oddball, AM, IDs]
+  in_names: [std, ba, wa,
+             tone,
+             s1, s2, s3, s4, s5]
+  in_numbers: [103, 104, 105,
+               1003,
+               10001, 10002, 10003, 10004, 10005]
+  out_names: [[All],
+              [standard, ba, wa],
+              [standard, deviant],
```
Comment (ktavabi) [inline badbaby/processing/badbaby.yml]: Ln436-437 requires matching.

File: badbaby/processing/score.py [hunk 5]
```diff
@@ -51,10 +52,11 @@ def score(p, subjects):
             fname_out = op.join(out_dir, f"ALL_{run_name % subj}-eve.lst")
 
             events, _ = extract_expyfun_events(fname)[:2]
-            events[:, 2] += 100
             if run_name in ("%s_am", "%s_ids"):
+                events[:, 2] += offsets[run_name.split('_')[-1]]
```
Comment (ktavabi) [inline badbaby/processing/score.py]: off course it was that easy 😔

File: badbaby/processing/badbaby.yml [hunk 6]
```diff
@@ -150,174 +79,131 @@ fetch_raw:
...
-      bad_312a: ['MEG1421, MEG1431'],
-      bad_312b: ['MEG1041'],
-      bad_314a: ['MEG1413'],
-      bad_316b: [],
-      bad_921a: [],
-      bad_921b: [],
-      bad_923b: ['MEG1823'],
-      bad_925b: []
-     }
+    mf_prebad:
```
Comment (ktavabi) [inline badbaby/processing/badbaby.yml:128]: I know you didn't ctrl+c ctrl+v to get this dictionary...did you?

**Review by ktavabi — APPROVED** [review]
This looks bute! apart from questions I will merge.

File: badbaby/processing/badbaby.yml (hunk 6 above)
Comment (larsoner) [reply badbaby/processing/badbaby.yml:128]: No it was tedious manual inspection for flux jumps and noise...

File: badbaby/processing/badbaby.yml (hunk 2 above)
Comment (larsoner) [reply badbaby/processing/badbaby.yml:223]: the filtering parameters should just be better overall, this st_correlation is safer / more conservative to make sure we aren't losing meaningful components

File: badbaby/processing/badbaby.yml (hunk 3 above)
Comment (larsoner) [reply badbaby/processing/badbaby.yml]: Needs to match the len of analyses

**Pull request merged**